helix-core/src/main/java/org/apache/helix/HelixCloudProperty.java

```diff
@@ -0,0 +1,150 @@
+    switch (cloudConfig.getCloudProvider()) {
+      case "AZURE":
+        Properties azureProperties = new Properties();
+        try {
+          InputStream stream =
+              Thread.currentThread().getContextClassLoader().getResourceAsStream(AZURE_CLOUD_PROPERTY_FILE);
+          azureProperties.load(stream);
+        } catch (Exception e) {
+          String errMsg = "fail to open properties file: " + AZURE_CLOUD_PROPERTY_FILE;
+          throw new IllegalArgumentException(errMsg, e);
+        }
+        logger.info("load helix Azure cloud properties: " + azureProperties);
+        this.setCloudInfoSources(Collections.singletonList(azureProperties.getProperty("CLOUD_INFO_SOURCE")));
+        this.setCloudInfoProcessorName(azureProperties.getProperty("CLOUD_INFO_PROCESSFOR_NAME"));
+        this.setCloudMaxRetry(azureProperties.getProperty("RETRY_MAX"));
+        this.setCloudConnectionTimeout(azureProperties.getProperty("CONNECTION_TIMEOUT_MS"));
+        this.setCloudRequestTimeout(azureProperties.getProperty("REQUEST_TIMEOUT_MS"));
+      case "CUSTOMIZED":
+        setCloudInfoSources(cloudConfig.getCloudInfoSources());
+        this.setCloudInfoProcessorName(cloudConfig.getCloudInfoProcessorName());
+    }
+  }
+
+  public Boolean getCloudEnabled() {
+    return _isCloudEnabled;
+  }
+
+  public String getCloudId() {
+    return _cloudId;
+  }
+
+  public String getCloudProvider() {
+    return _cloudProvider;
+  }
+
+  public List<String> getCloudInfoSources() {
+    return _cloudInfoSources;
+  }
+
+  public String getCloudInfoProcessorName() {
+    return _cloudInfoProcessorName;
+  }
+
+  public String getMaxRetry() {
+    return _maxRetry;
+  }
+
+  public String getCloudConnectionTimeout() {
+    return _cloudConnectionTimeout;
+  }
+
+  public String getCloudRequestTimeout() {
+    return _cloudRequestTimeout;
+  }
+
+  public Properties getCustomizedProperties() {
+    return _customizedProperties;
+  }
+
+  public void setCloudEndabled(Boolean isCloudEnabled) {
+    _isCloudEnabled = isCloudEnabled;
+  }
+
+  public void setCloudId(String cloudId) {
+    _cloudId = cloudId;
+  }
+
+  public void setCloudProvider(String cloudProvider) {
+    _cloudProvider = cloudProvider;
+  }
+
+  public void setCloudInfoSources(List<String> sources) {
+    _cloudInfoSources = sources;
+  }
+
+  public void setCloudInfoProcessorName(String cloudInfoProcessorName) {
+    _cloudInfoProcessorName = cloudInfoProcessorName;
+  }
+
+  public void setCloudMaxRetry(String maxRetry) {
+    _maxRetry = maxRetry;
+  }
+
+  public void setCloudConnectionTimeout(String cloudConnectionTimeout) {
+    _cloudConnectionTimeout = cloudConnectionTimeout;
+  }
+
+  public void setCloudRequestTimeout(String cloudRequestTimeout) {
+    _cloudRequestTimeout = cloudRequestTimeout;
+  }
+
+  public void setCustomizedCloudProperties(Properties customizedCloudProperties) {
+    _customizedProperties.putAll(customizedCloudProperties);
+  }
```

Comment:
Ideally every public method/API should have a pertinent JavaDoc to go with explaining what they do and what these strings look like (examples)